api/src/dto/ChangeDto.ts

```diff
@@ -1,21 +1,39 @@
-import { DocType, Uuid } from "../types";
-import { IsNotEmpty, IsObject } from "class-validator";
+import { DocType, Uuid } from "../enums";
+import {
+    IsArray,
+    IsEnum,
+    IsNotEmpty,
+    IsObject,
+    IsOptional,
+    IsString,
+    ValidateNested,
+} from "class-validator";
 import { GroupAclEntryDto } from "./GroupAclEntryDto";
+import { Type } from "class-transformer";
+import { _baseDto } from "./_baseDto";
 
 /**
  * Database structured Post object
  */
-export class ChangeDto {
-    @IsNotEmpty()
-    _id: Uuid;
-    @IsNotEmpty()
-    type: DocType.Change;
+export class ChangeDto extends _baseDto {
     @IsNotEmpty()
+    @IsString()
     docId: Uuid;
+
     @IsNotEmpty()
+    @IsEnum(DocType)
     docType: DocType;
+
+    @IsOptional()
+    @IsArray()
+    @IsString({ each: true })
     memberOf?: Uuid[];
+
+    @IsOptional()
+    @ValidateNested({ each: true })
+    @Type(() => GroupAclEntryDto) // This throws an exception on validation failure, so we need to catch the error on validation. The message is less user-friendly but at least the validator fails and will protect our data.
```

Comment:
It's possible to customize the message which would make sense if it's not super clear (and especially if we want to display it in the frontend): https://github.com/typestack/class-validator?tab=readme-ov-file#validation-messages

Reply:
Throwing an exception makes it not show the message at all. So not very helpful here. Custom messages are however nice, and can be used when we add more detailed validations.